iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/parser/properties/PropertiesParseTreeVisitor.java

```diff
@@ -0,0 +1,105 @@
+import org.sonar.iac.springconfig.tree.api.Profile;
+import org.sonar.iac.springconfig.tree.api.Scalar;
+import org.sonar.iac.springconfig.tree.api.SpringConfig;
+import org.sonar.iac.springconfig.tree.api.Tuple;
+import org.sonar.iac.springconfig.tree.impl.FileImpl;
+import org.sonar.iac.springconfig.tree.impl.ProfileImpl;
+import org.sonar.iac.springconfig.tree.impl.ScalarImpl;
+import org.sonar.iac.springconfig.tree.impl.SyntaxTokenImpl;
+import org.sonar.iac.springconfig.tree.impl.TupleImpl;
+
+public class PropertiesParseTreeVisitor extends PropertiesParserBaseVisitor<SpringConfig> {
+
+  private static final List<String> PROFILE_NAME_PROPERTIES = List.of("spring.profile", "spring.config.active.on-profile");
+
+  @Override
+  public SpringConfig visitPropertiesFile(PropertiesParser.PropertiesFileContext ctx) {
+    super.visitPropertiesFile(ctx);
+    var properties = new ArrayList<Tuple>();
+    var comments = new ArrayList<Comment>();
+    var profiles = new ArrayList<Profile>();
+    for (PropertiesParser.RowContext row : ctx.row()) {
+      if (row.line() != null) {
+        var keyScalar = createKeyScalar(row);
+        var valueScalar = createValueScalar(row);
+        properties.add(new TupleImpl(keyScalar, valueScalar));
+      }
+      if (row.comment() != null) {
+        var commentContext = row.comment().commentStartAndText();
+        var value = commentContext.getText();
+        var contentText = commentContext.commentText().getText();
+        if ("---".equals(contentText)) {
+          var profile = new ProfileImpl(properties, comments, profileName(properties), true);
+          profiles.add(profile);
+          properties = new ArrayList<>();
+          comments = new ArrayList<>();
+        }
+        comments.add(new CommentImpl(value, contentText, textRange(commentContext)));
+      }
+    }
+    var profile = new ProfileImpl(properties, comments, profileName(properties), true);
+    profiles.add(profile);
+    return new FileImpl(profiles);
+  }
+
+  private static Scalar createKeyScalar(PropertiesParser.RowContext row) {
+    var keyContext = row.line().key().get(0);
+    var keyText = keyContext.getText();
+    return new ScalarImpl(new SyntaxTokenImpl(keyText, textRange(keyContext)));
+  }
+
+  private static Scalar createValueScalar(PropertiesParser.RowContext row) {
+    Scalar valueScalar = null;
+    if (row.line().key().size() == 2) {
+      var valueContext = row.line().key().get(1);
+      var valueText = valueContext.getText();
+      valueScalar = new ScalarImpl(new SyntaxTokenImpl(valueText, textRange(valueContext)));
+    }
+    return valueScalar;
+  }
+
+  private static String profileName(List<Tuple> properties) {
+    return properties.stream()
+      .filter(tuple -> PROFILE_NAME_PROPERTIES.contains(tuple.key().value().value()))
+      .filter(tuple -> tuple.value() != null)
+      .map(tuple -> tuple.value().value().value())
+      .collect(Collectors.joining(" "));
+  }
```

Comment:
Not sure about the concatenation, see other discussion.
Nevertheless, this method can be extracted into a separate class, because it makes sense that I use it too and don't implement the same logic again)
(Or I will extract it later)

Reply:
I'm also not sure and we should validate it later and adapt.